tests/BiosTests.hs

```diff
@@ -195,15 +206,38 @@ findCradleForModule fp expected' step = do
   step "Finding cradle"
   findCradle a_fp `shouldReturn` expected
 
-testImplicitCradle :: FilePath -> ActionName Void -> (String -> IO ()) -> IO ()
-testImplicitCradle fp' expectedActionName step = do
-  fp <- canonicalizePath fp'
+testImplicitCradle :: FilePath -> ActionName Void -> FilePath -> (String -> IO ()) -> IO ()
+testImplicitCradle fp' expectedActionName expectedCradleRootDir step = do
+  fp <- makeAbsolute fp'
   step "Inferring implicit cradle"
   crd <- loadImplicitCradle fp :: IO (Cradle Void)
+
   actionName (cradleOptsProg crd) `shouldBe` expectedActionName
+
+  absExpectedCradleRootDir <- makeAbsolute expectedCradleRootDir
+  cradleRootDir crd `shouldBe` absExpectedCradleRootDir
+
   step "Initialize flags"
   testLoadFile crd fp step
 
+copyToTmp :: FilePath -> (FilePath -> IO a) -> IO a
+copyToTmp srcDir f =
```

Comment:
Nitpick, I dont like the name (as always, this opinion can be debated)
I think I prefer something like `runTestInTmpDir` or maybe even `runTestInTemporaryDirectory`